### PR TITLE
Prediction: get rid of perception message in evaluator and predictor

### PR DESCRIPTION
**Pull request by kechxu**

### CODE REVIEW

File: modules/prediction/container/obstacles/obstacles_container.h [hunk 1]
```diff
@@ -102,6 +102,8 @@ class ObstaclesContainer : public Container {
 
   size_t NumOfObstacles() { return ptr_obstacles_.size(); }
 
+  apollo::perception::PerceptionObstacle GetPerceptionObstacle(const int id);
+
```
Comment (YajiaZhang) [inline modules/prediction/container/obstacles/obstacles_container.h]: Better rename "id" to "perception_obstacle_id"

File: modules/prediction/container/obstacles/obstacles_container.cc [hunk 2]
```diff
@@ -169,13 +174,16 @@ void ObstaclesContainer::InsertPerceptionObstacle(
...
+  curr_frame_id_perception_obstacle_map_[prediction_id] = perception_obstacle;
   if (prediction_id < -1) {
     AERROR << "Invalid ID [" << prediction_id << "]";
     return;
   }
   if (!IsPredictable(perception_obstacle)) {
     ADEBUG << "Perception obstacle [" << prediction_id
            << "] is not predictable.";
+    curr_frame_non_predictable_obstacle_ids_.push_back(
+        perception_obstacle.id());
```
Comment (YajiaZhang) [inline modules/prediction/container/obstacles/obstacles_container.cc]: Here, the "id" is prediction_obstacle_id or "perception_obstacle_id"? It looks like the two are mixed.

Comment (kechxu) [reply modules/prediction/container/obstacles/obstacles_container.cc]: Thanks, I will fix it.

File: modules/prediction/container/obstacles/obstacles_container.cc [hunk 3]
```diff
@@ -41,6 +41,10 @@ void ObstaclesContainer::Insert(const ::google::protobuf::Message& message) {
   // Clean up the history and get the PerceptionObstacles
   curr_frame_predictable_obstacle_ids_.clear();
   curr_frame_id_mapping_.clear();
+  curr_frame_predictable_obstacle_ids_.clear();
```
[the remark this answers is not in the document]
Comment (kechxu) [reply modules/prediction/container/obstacles/obstacles_container.cc]: Thanks, will fix it.

File: modules/prediction/predictor/predictor_manager.cc [hunk 4]
```diff
@@ -137,25 +136,20 @@ void PredictorManager::Run(const PerceptionObstacles& perception_obstacles) {
...
   Predictor* predictor = nullptr;
-  for (const auto& perception_obstacle :
-       perception_obstacles.perception_obstacle()) {
-    if (!perception_obstacle.has_id()) {
-      AERROR << "A perception obstacle has no id.";
-      continue;
-    }
-
-    int id = perception_obstacle.id();
+  for (int id : obstacles_container->curr_frame_obstacle_ids()) {
```
[the remark this answers is not in the document]
Comment (kechxu) [reply modules/prediction/predictor/predictor_manager.cc]: Thanks, will fix it.

File: modules/prediction/container/obstacles/obstacles_container.cc [hunk 5]
```diff
@@ -126,19 +127,23 @@ Obstacle* ObstaclesContainer::GetObstacleWithLRUUpdate(const int obstacle_id) {
...
   id_mapping_.Clear();
   timestamp_ = -1.0;
 }
 
+PerceptionObstacle ObstaclesContainer::GetPerceptionObstacle(const int id) {
+  CHECK(curr_frame_id_perception_obstacle_map_.find(id) !=
+        curr_frame_id_perception_obstacle_map_.end());
+  return curr_frame_id_perception_obstacle_map_[id];
+}
+
```
Comment (YajiaZhang) [inline modules/prediction/container/obstacles/obstacles_container.cc]: Will "PerceptionObstacle" be changed inside prediction module? If not, it is much more efficient to return const ptr. You can modify this in next PR.

**Pull request merged**